### PR TITLE
Intel compiler support

### PR DESCRIPTION
**Pull request by egecetin**

### CODE REVIEW

File: .github/workflows/build_and_test.yml
Comment (seladb) [inline .github/workflows/build_and_test.yml:52]: maybe we can add another variable `additional-flags: -DCMAKE_CXX_COMPILER=icpx` and then add it in the `Configure PcapPlusPlus` step?

```yaml
run: cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -DPCAPPP_BUILD_COVERAGE=ON ${{additional-flags}} -S . -B $BUILD_DIR
```

**Review by seladb — APPROVED** [review]

**Comment by egecetin** [comment]
@seladb If CI passes after merge of this [PR](https://github.com/seladb/PcapPlusPlus-DockerImages/pull/24) it is ok i think. Can you take a look again?

**Comment by seladb** [comment]
> @seladb If CI passes after merge of this [PR](https://github.com/seladb/PcapPlusPlus-DockerImages/pull/24) it is ok i think. Can you take a look again?

Sure @egecetin , I think we can merge this one after the dockerfile PR is merged

**Pull request merged**
